Example/ViewController.swift

```diff
@@ -82,133 +99,188 @@ final class ViewController: UIViewController {
+        case .currentTime(let time, _):
+            guard let t = time else { return nil }
+            return String(format: "%.2f", t)
+        case .seekRatio(let ratio):
+            return "ratio:\(ratio)"
+        case .seekTime(let time):
+            return String(format: "%.2f", time)
+        }
     }
     
-    func modernAVPlayer(_ player: ModernAVPlayer, didCurrentTimeChange currentTime: Double?) {
-        self.currentTime = currentTime
+    private func createPositionRequest(state: ModernAVPlayer.State,
+                                       currentTime: Double?,
+                                       isSliderSeeking: Bool,
+                                       sliderPosition: Float,
+                                       optDuration: Double?) -> PositionRequest {
+        
+        guard isSliderSeeking || state == .buffering
+            else { return .currentTime(time: currentTime, duration: optDuration) }
+        
+        guard let duration = optDuration else { return .seekRatio(sliderPosition) }
+        
+        return .seekTime(Double(sliderPosition) * duration)
     }
     
-    func modernAVPlayer(_ player: ModernAVPlayer, didItemDurationChange itemDuration: Double?) {
-        self.itemDuration = itemDuration
+    private func setSliderPosition(current: Double, duration: Double) -> Float {
+        return Float(current / duration)
     }
     
-    func modernAVPlayer(_ player: ModernAVPlayer, debugMessage: String?) {
-        setDebugMessage(debugMessage)
+    private func setSeekPosition(from duration: Double) -> Double {
+        return Double(slider.value) * duration
     }
+}
+
+// MARK: - Player RxSwift
+
+extension ViewController {
     
-    func modernAVPlayer(_ player: ModernAVPlayer, didCurrentItemUrlChange currentItemUrl: URL?) {
-        DispatchQueue.main.async {
-            self.currentItemLabel.text = currentItemUrl?.absoluteString
-        }
+    private func bindPlayerRxAttibutes() {
+        
+        // Display current time label
+        Observable
+            .combineLatest(player.rx.currentTime.distinctUntilChanged(),
+                           slider.rx.value.asObservable().distinctUntilChanged())
+            .withLatestFrom(itemDurationSubject.asObservable()) { return ($0.0, $0.1, $1) }
+            .withLatestFrom(player.rx.state) { return ($0.0, $0.1, $0.2, $1) }
+            .withLatestFrom(isSliderSeeking) { return ($0.3, $0.0, $1, $0.1, $0.2) }
+            .subscribeOn(concurrentBackgroundScheduler)
+            .map(createPositionRequest)
+            .map(formatPosition)
+            .asDriver(onErrorJustReturn: "error")
+            .drive(timingLabel.rx.text)
+            .disposed(by: disposeBag)
+        
+        // Seek
+        isSliderSeeking
+            .distinctUntilChanged()
+            .skip(1) // skip init value
+            .filter { !$0 }
+            .withLatestFrom(itemDurationSubject.asObservable())
+            .filter { $0 != nil }.map { $0! }
```

Comment:
missing CRLF

Reply:
I dit it intentionally 
.filter { $0 != nil }.map { $0! } = unwrap

more readable for me, but maybe not

Reply:
Hum I see!
you can also do:
```swift 
            .flatMap { Observable.from(optional: $0) }
```
your choice. I'm ok with each

Reply:
didn't know this way, thanks.